linkerd/retry/src/lib.rs

```diff
@@ -18,29 +20,9 @@ pub trait NewPolicy<T> {
     fn new_policy(&self, target: &T) -> Option<Self::Policy>;
 }
 
-/// A layer that applies per-target retry policies.
-///
-/// Composes `NewService`s that produce a `Proxy`.
-#[derive(Clone, Debug)]
-pub struct NewRetryLayer<P> {
-    new_policy: P,
-}
-
-#[derive(Clone, Debug)]
-pub struct NewRetry<P, N> {
-    new_policy: P,
-    inner: N,
-}
-
-#[derive(Clone, Debug)]
-pub struct Retry<P, S> {
-    policy: Option<P>,
-    inner: S,
-}
-
 /// An extension to [`tower::retry::Policy`] that adds a method to prepare a
 /// request to be retried, possibly changing its type.
-pub trait RetryPolicy<Req, Res, E>: tower::retry::Policy<Self::RetryRequest, Res, E> {
+pub trait PrepareRequest<Req, Res, E>: tower::retry::Policy<Self::RetryRequest, Res, E> {
```

Comment:
my original motivation for naming this `RetryPolicy` --- which i agree is a bit confusing --- was because it requires `tower`'s `RetryPolicy` trait, so when it's used, we only bound the type with _this_ trait. 

i also considered the name `PrepareRequest`, but it seemed kind of weird to me to require that the policy implement `PrepareRequest` and not mention that it is also a retry policy? maybe that's not actually worth worrying about, and i'm fine with this naming scheme if you think it's clearer...

Reply:
could call it `PreparePolicy` or something like that -- but RetryPolicy vs Policy is a bit too subtle imo